mathesar/functions/operations/convert.py

```diff
@@ -0,0 +1,44 @@
+from db.functions.operations.deserialize import get_raw_spec_components
+from db.functions.exceptions import ReferencedColumnsDontExist
+
+
+def rewrite_db_function_spec_column_ids_to_names(column_ids_to_names, spec):
+    """
+    Takes a DB function spec, looks for columns referenced via Django IDs (e.g. `{"column_id": 3}`)
+    and replaces those with column names. That's necessary since the DB module is unaware of column
+    IDs and can only accept columns referenced by name.
+    """
+    # Using a private method to do the heavy lifting since it uses different parameters.
+    return _rewrite(
+        column_ids_to_names=column_ids_to_names,
+        spec_or_literal=spec
+    )
+
+
+def _rewrite(column_ids_to_names, spec_or_literal):
```

Comment:
I only see the JSON spec being parsed here - couldn't you check for single parameters here and convert them to a list during this rewrite?

Reply:
The other place is `db.functions.operations.deserialize`.